astropy/utils/data.py

```diff
@@ -1259,9 +1280,15 @@ def clear_download_cache(hashorurl=None):
                 elif hash_key in url2hash:
                     filepath = url2hash[hash_key]
                     del url2hash[hash_key]
-                    if os.path.exists(filepath):
-                        # Make sure the filepath still actually exists (perhaps user removed it)
-                        os.unlink(filepath)
+                    for k, v in url2hash.items():
```

Comment:
I think it would be clearer to write
```
# Check nothing else points here and that
# the filepath still exists
if (not any(v == filepath for v in url2hash.values()) and
        os.path.exists(filepath)):
    os.unlink(filepath)
```